edge-modules/TwinTester/TwinOperationBase.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TwinTester
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Extensions.Logging;
+
+    public abstract class TwinOperationBase
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger(nameof(TwinOperationBase));
+
+        public abstract Task PerformUpdateAsync();
+
+        public abstract Task PerformValidationAsync();
+
+        protected static bool IsPastFailureThreshold(TwinState twinState, DateTime twinUpdateTime)
```

Comment:
These static methods look like helper methods, I think they should not be in the base class. You could have a TwinOperationHelper class and the current TwinOperationBase change it to an interface

Reply:
Philip and I took away static so now they are members of the abstract class.